### PR TITLE
Updated detect_machine.sh to be consistent with UFS WM.

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
This PR addresses issue #41. The following is accomplished:

- the `detect_machine.sh` script is updated to be consistent with that of the UFS WM.

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
looks good.
thanks.

File: modulefiles/gfsutils_hera.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
+help([[
```
Comment (aerorahul) [inline modulefiles/gfsutils_hera.lua]: I don't think this is needed. Instead, the other modulefiles need to be renamed to include the `COMPILER` in their name similar to the `gfsutils_hera.intel.lua` name.

**Review by aerorahul — CHANGES_REQUESTED** [review]
please see a comment inline

File: ush/detect_machine.sh [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+# The authoritative copy of this script lives in the ufs-weather-model at:
+# https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/detect_machine.sh
+# If any local modifications are made or new platform support added,
+# please consider opening an issue and a PR to the ufs-weather-model
+# so that this copy remains in sync with its authoritative source
+#
+# Thank you for your contribution
+
+# If the MACHINE_ID variable is set, skip this script.
+[[ -n ${MACHINE_ID:-} ]] && return
```
Comment (WalterKolczynski-NOAA) [inline ush/detect_machine.sh:12]: This should probably properly be `-v MACHINE_ID`, but since the goal is to match UFS, it's fine.

**Review by aerorahul — APPROVED** [review]
thanks for updating the modulefiles to be consistent.

**Pull request merged**